tools/mini-yetus.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Do you hate yetus like everyone else? well I can't do anything about that but
+# using this you can at least hate `make yetus` a little less. This script will
+# run yetus on the changes in your current branch compared to master (excluding
+# vendor changes) and output the results, much faster than running it on the
+# entire codebase with `make yetus`.
+
+echo "[+] Running mini-yetus"
+# check if we are in the root of the repository
+if [ ! -d .git ]; then
+    echo "[!] Error: This script must be run from the root of the EVE repository."
+    exit 1
+fi
+
+if [ "$#" -eq 1 ]; then
+    DST_BRANCH="$1"
+else
+    # No branch specified, use the current branch
+    DST_BRANCH=$(git rev-parse --abbrev-ref HEAD 2>/dev/null)
+    if [ -z "$DST_BRANCH" ]; then
+        echo "[!] Error: Could not determine the current branch."
+        exit 1
+    fi
+    echo "[+] No branch specified. Using the current branch: $DST_BRANCH"
+fi
+
+if ! git rev-parse --verify "$DST_BRANCH" >/dev/null 2>&1; then
+    echo "[!] Error: Branch '$DST_BRANCH' does not exist."
+    exit 1
+fi
+
+SRC_DIR=$(mktemp -d)
+git diff --name-only master.."$DST_BRANCH" | grep -v '/vendor/' | while IFS= read -r file; do
+    if [ -e "$file" ]; then
+        cp --parents -r "$PWD/$file" "$SRC_DIR/" || { echo "Failed to copy $file"; exit 1; }
+    else
+        echo "[!] File does not exist: $PWD/$file"
+        exit 1
+    fi
+done
+
+# copy all the dot files from root to the SRC_DIR, this includes all
+# the yetus configuration files.
+find . -maxdepth 1 -type f -name '.*' -exec cp --parents {} "$SRC_DIR/" \;
+
+cd "$SRC_DIR" && \
+    git init >/dev/null 2>&1 && \
+    git add . >/dev/null 2>&1 && \
+    git config user.email "you@example.com" >/dev/null 2>&1 && \
+    git config user.name "Your Name" >/dev/null 2>&1 && \
+    git commit -m "Changes in the PR" >/dev/null 2>&1
+
+echo "[+] Running yetus on the changes..."
+docker run --rm -v "$SRC_DIR":/src:delegated,z ghcr.io/apache/yetus:0.15.0 \
+    --basedir=/src \
+    --test-parallel=true \
+    --dirty-workspace \
+    --empty-patch \
+    --plugins=all \
+    --patch-dir=/src/yetus-output > /dev/null 2>&1
+
+echo "[+] Results:"
+cat "$SRC_DIR/yetus-output/results-full.txt"
```

Comment:
IMHO we should let the users decide to which files they want to look at it and remove the output by themselves... in my case I go directly to the result file I'm interested, e.g., results-shellcheck.txt or results-golangcilint.txt, here it's enforced to always get the full results and remove the output dir, sometimes users can also want to keep the results saved....

Reply:
sure.